tools/pkglint/lib/rules.ts

```diff
@@ -46,9 +46,32 @@ export class DescriptionIsRequired extends ValidationRule {
   }
 }
 
+/**
+ * Verify that all packages have a publishConfig with a publish tag set.
+ */
+export class PublishConfigTagIsRequired extends ValidationRule {
+  public readonly name = 'package-info/publish-config-tag';
+
+  public validate(pkg: PackageJson): void {
+    const defaultPublishTag = (cdkMajorVersion() === 2) ? 'next' : 'latest';
+
+    if (!pkg.json.publishConfig?.tag) {
+      pkg.report({
+        ruleName: this.name,
+        message: 'publishConfig.tag is required',
+        fix: (() => {
+          const publishConfig = pkg.json.publishConfig ?? {};
+          publishConfig.tag = defaultPublishTag;
+          pkg.json.publishConfig = publishConfig;
+        }),
+      });
```

Comment:
Don't we need special logic for `aws-cdk-lib`, i.e., `latest` instead of `next`?

Reply:
We _could_ encode it here, but I was planning on just making the change in the `aws-cdk-lib` package itself. The auto-fix is here to automatically adjust packages where the tag is missing entirely, not encode the special one-off rule for the single exception (IMO).

Reply:
I think it's better to have all of the logic encoded here. We should also err/autofix if the tag is incorrect, not just for missing.